### PR TITLE
add reduction of sum and  none for CrossEntropyLoss 

### PR DESCRIPTION
**Pull request by FatJhon**
1. modify inaccurate variable names
2. add reduction of sum and none for CrossEntropyLoss

### CODE REVIEW

File: src/flag_gems/ops/cross_entropy_loss.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def log_softmax_and_mul_kernel(
     denominator = tl.sum(numerator, axis=1)[:, None]
     softmax_output = tl.log(numerator / denominator)
     target = tl.load(target_ptr + offset, mask=mask, other=0.0)
-    out = softmax_output * target / (-mean_num)
+    out = softmax_output * target / (mean_num)
```
Comment (tongxin) [inline src/flag_gems/ops/cross_entropy_loss.py:66]: Can we unify these fused log softmax and blah functions

File: src/flag_gems/ops/cross_entropy_loss.py [hunk 2]
```diff
@@ -130,10 +192,12 @@ def forward(ctx, input, target, weight, reduction, ignore_index, label_smoothing
             dim = 1
         else:
             dim = 0
-
+        if reduction != 1:
```
Comment (tongxin) [inline src/flag_gems/ops/cross_entropy_loss.py]: Can we explicitly define an enum type for reduction?

File: src/flag_gems/ops/cross_entropy_loss.py [hunk 3]
```diff
@@ -157,11 +221,15 @@ def forward(ctx, input, target, weight, reduction, ignore_index, label_smoothing
             N,
             K,
         )
-        out_result = sum(out)
+        if reduction:
```
Comment (tongxin) [inline src/flag_gems/ops/cross_entropy_loss.py]: Please be explicit on the reduction types

File: src/flag_gems/ops/cross_entropy_loss.py (hunk 1 above)
Comment (FatJhon) [reply src/flag_gems/ops/cross_entropy_loss.py:66]: I will try. The difference is that when reduction is none，out_grad is a tensor,  reduction is sum or mean, out_grad is a constant, it is difficult to separate in triton.jt

File: src/flag_gems/ops/cross_entropy_loss.py (hunk 3 above)
Comment (FatJhon) [reply src/flag_gems/ops/cross_entropy_loss.py]: done

File: src/flag_gems/ops/cross_entropy_loss.py (hunk 2 above)
Comment (FatJhon) [reply src/flag_gems/ops/cross_entropy_loss.py]: done

File: src/flag_gems/ops/cross_entropy_loss.py (hunk 1 above)
Comment (tongxin) [reply src/flag_gems/ops/cross_entropy_loss.py:66]: Got it.

File: src/flag_gems/ops/cross_entropy_loss.py [hunk 4]
```diff
@@ -2,8 +2,15 @@
 import triton
 import triton.language as tl
 import logging
+from enum import Enum
 from ..utils import libentry
-from .sum import sum
+from .sum import sum, sum_dim
+
+
+class Reduction(Enum):
```
Comment (tongxin) [inline src/flag_gems/ops/cross_entropy_loss.py]: Use IntEnum then its members are also integers.

Comment (FatJhon) [reply src/flag_gems/ops/cross_entropy_loss.py]: Thanks  for your correction. The interface usage will be more precise.
